components/suggest/src/db.rs

```diff
@@ -186,7 +186,9 @@ impl<'a> SuggestDao<'a> {
             .iter()
             .try_fold(vec![], |mut acc, provider| {
                 let suggestions = match provider {
-                    SuggestionProvider::Amp => self.fetch_amp_suggestions(query),
+                    SuggestionProvider::Amp | SuggestionProvider::AmpMobile => {
+                        self.fetch_amp_suggestions(query, **provider)
+                    }
```

Comment:
So I tried my best to recycle the existing bits as much as I can, but as a result, this look a little goofy since at the point of `fetch_amp_suggestions` we don't know which provider it came from.